probes/securityPolicyContainsLinks/impl.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:stylecheck
+package securityPolicyContainsLinks
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/internal/utils"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const Probe = "securityPolicyContainsLinks"
+
+func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
+	if raw == nil {
+		return nil, "", fmt.Errorf("%w: raw", utils.ErrNil)
+	}
+	var findings []finding.Finding
+	policies := raw.SecurityPolicyResults.PolicyFiles
+	for i := range policies {
+		policy := &policies[i]
+		emails := utils.CountSecInfo(policy.Information, checker.SecurityPolicyInformationTypeEmail, true)
+		urls := utils.CountSecInfo(policy.Information, checker.SecurityPolicyInformationTypeLink, true)
+
+		if (urls + emails) > 0 {
+			f, err := finding.NewPositive(fs, Probe,
+				"URL or email contacts found", policy.File.Location())
+			if err != nil {
+				return nil, Probe, fmt.Errorf("create finding: %w", err)
+			}
+			findings = append(findings, *f)
+		} else {
+			f, err := finding.NewNegative(fs, Probe,
+				"no URL or email contacts found", nil)
+			if err != nil {
+				return nil, Probe, fmt.Errorf("create finding: %w", err)
+			}
+			findings = append(findings, *f)
+		}
+	}
+
+	if len(findings) == 0 {
+		f, err := finding.NewNegative(fs, Probe, "no security file to analyze", nil)
+		if err != nil {
+			return nil, Probe, fmt.Errorf("create finding: %w", err)
+		}
+		findings = append(findings, *f)
```

Comment:
I think we can get some easy test coverage for all 4 probes if we set up a `impl_test.go` file which calls the probe. Maybe one test with a nil raw data to check for error. And one test with an empty raw data to check for negative probe.

Reply:
Tests under evaluation exercise this code. But we can add them here too

Reply:
I believe the tests under evaluation assume the probes have already been run, and try to turn it into a score.

There are tests under checks/security_policy_test.go, but they all have a file they're testing. 

I think this logic is actually untested

Reply:
wrong, the tests under evaluation have the findings already set up